src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -113,127 +135,378 @@ internal abstract class AbstractAddParameterCodeFixProvider<
             }
 
             var arguments = (SeparatedSyntaxList<TArgumentSyntax>)syntaxFacts.GetArgumentsOfObjectCreationExpression(objectCreation);
+            var methodCandidates = type.InstanceConstructors;
+
+            var insertionData = GetArgumentInsertPositionForMethodCandidates(
+                argumentOpt, semanticModel, syntaxFacts, arguments, methodCandidates);
+
+            RegisterFixForMethodOverloads(context, arguments, insertionData);
+        }
 
+        private ImmutableArray<ArgumentInsertPositionData<TArgumentSyntax>> GetArgumentInsertPositionForMethodCandidates(
+            TArgumentSyntax argumentOpt,
+            SemanticModel semanticModel,
+            ISyntaxFactsService syntaxFacts,
+            SeparatedSyntaxList<TArgumentSyntax> arguments,
+            ImmutableArray<IMethodSymbol> methodCandidates)
+        {
             var comparer = syntaxFacts.StringComparer;
-            var constructorsAndArgumentToAdd = ArrayBuilder<(IMethodSymbol constructor, TArgumentSyntax argument, int index)>.GetInstance();
+            var methodsAndArgumentToAdd = ArrayBuilder<ArgumentInsertPositionData<TArgumentSyntax>>.GetInstance();
 
-            foreach (var constructor in type.InstanceConstructors.OrderBy(m => m.Parameters.Length))
+            foreach (var method in methodCandidates.OrderBy(m => m.Parameters.Length))
             {
-                if (constructor.IsNonImplicitAndFromSource() &&
-                    NonParamsParameterCount(constructor) < arguments.Count)
+                if (method.IsNonImplicitAndFromSource())
                 {
-                    var argumentToAdd = DetermineFirstArgumentToAdd(
-                        semanticModel, syntaxFacts, comparer, constructor,
-                        arguments, argumentOpt);
+                    var isNamedArgument = !string.IsNullOrWhiteSpace(syntaxFacts.GetNameForArgument(argumentOpt));
 
-                    if (argumentToAdd != null)
+                    if (isNamedArgument || NonParamsParameterCount(method) < arguments.Count)
                     {
-                        if (argumentOpt != null && argumentToAdd != argumentOpt)
+                        var argumentToAdd = DetermineFirstArgumentToAdd(
```

Comment:
would it be possible extending it to add multiple arguments? I saw one of your tests that new multiple new arguments and only one of that got added. I guess in such situation, users need to ctrl+. and apply fix and then repeat it until all new arguments are added. but it would be nice if that can be done 1 fix.

Reply:
This fixer depends on diagnostic errors and the diagnostic is only reported for one argument. I could imagine all kinds of trouble if there is more than one overload present. I still like the idea but I would suggest to fix this in another PR later. There are also some other ideas around generic parameters that could also be done later in other PRs (maybe based on customer feedback).